### ISSUE TITLE
Inference in 01_ASR_with_NeMo fails to put model in eval mode

### ISSUE THREAD

**Issue by rbracco** [NVIDIA/NeMo#1533]
**Describe the bug**

The inference code in the ASR tutorial 01_ASR_with_NeMo doesn't put the model in training mode before inference. If copied and used in someone's code, this would cause augmentation like SpecAugment to be inadvertantly be applied to the test set resulting in higher WER than the model actually produces using `transcribe`. 

**Expected behavior**

The inference loop should 
1. Store the current mode (as transcribe does) with `mode=first_asr_model.training` 
2. Call `first_asr_model.eval()` before running the pytorch inference loop
3. Reset the mode with `first_asr_model.train(mode=mode)` upon completion

**Comment by titu1994** [comment]
Fixed in main branch, thanks for pointing it out !
